### PR TITLE
Jepsen/lock

### PR DESCRIPTION
**Pull request by gmaretic**
Added checkers for Lock and refactored some of the existing code

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1473)

### CODE REVIEW

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/JepsenHistoryChecker.java [hunk 1]
```diff
@@ -42,6 +43,11 @@ public JepsenHistoryChecker(List<Checker> checkers) {
         return checkers;
     }
 
+    public static JepsenHistoryChecker createWithStandardCheckersForLock() {
+        return new JepsenHistoryChecker(
+                new LockCorrectnessChecker());
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/JepsenHistoryChecker.java]: nit: the constructors that accept arbitrary checkers as input are public - do we really need this method?

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java [hunk 2]
```diff
@@ -32,15 +33,35 @@
...
+    public boolean isFailure() {
+        return value() == null
+                || (function().equals(RequestType.UNLOCK) && value().equals(UNLOCK_FAILURE))
+                || value().equals(FAILURE);
+    }
+
+    public boolean isSuccessful() {
+        return value().equals(SUCCESS)
+                || (function().equals(RequestType.UNLOCK) && value().equals(UNLOCK_SUCCESS));
+    }
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java]: These 2 checks seem rather specific to `OkEvent` interpreted _in the context of a lock checker_. I think it would be preferable to have these as static methods in the specific lock checker, or in some utility class that can be used by lock checkers, rather than being a property of the event.

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.jepsen.CheckerResult;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.utils.TestEventUtils;
+
+public class IsolatedProcessCorrectnessCheckerTest {
+    private static final int process1 = 1;
```
Comment (harrybiddle) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java]: Should be upper case to fit into checkstyle

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    public void cannotRefreshWithoutLock() {
+        ImmutableList<Event> eventList = ImmutableList.<Event>builder()
+                .add(TestEventUtils.invokeRefresh(0, process1))
+                .add(TestEventUtils.refreshSuccess(1, process1))
+                .add(TestEventUtils.invokeRefresh(2, process1, "alternate_lock"))
+                .add(TestEventUtils.refreshFailure(3, process1))
+                .build();
+        CheckerResult result = runIsolatedProcessRefreshSuccessChecker(eventList);
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(eventList.get(1), eventList.get(1));
```
Comment (harrybiddle) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java]: is this right? we expect to see the same event twice?

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.jepsen.CheckerResult;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.utils.TestEventUtils;
+
+public class LockCorrectnessCheckerTest {
```
Comment (harrybiddle) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java]: I know that we're potentially going to blow up the number of tests, but I think it's worth exercising the code that handles different lock names, by using multiple locks in at least some places here.

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/NemesisResilienceCheckerTest.java [hunk 6]
```diff
@@ -35,67 +34,31 @@
...
+            .createInfoEvent(ZERO_TIME, JepsenConstants.NEMESIS_PROCESS, JepsenConstants.STOP_FUNCTION, VALUE_1);
+
+    private static final Event NEMESIS_STOP_2 = TestEventUtils
+            .createInfoEvent(ZERO_TIME, JepsenConstants.NEMESIS_PROCESS, JepsenConstants.STOP_FUNCTION, VALUE_2);
+
+    private static final Event IMPOSTOR_START = TestEventUtils
+            .createInfoEvent(ZERO_TIME, IMPOSTOR_PROCESS, JepsenConstants.START_FUNCTION);
+
+    private static final Event IMPOSTOR_STOP = TestEventUtils
+            .createInfoEvent(ZERO_TIME, IMPOSTOR_PROCESS, JepsenConstants.STOP_FUNCTION);
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/NemesisResilienceCheckerTest.java]: Cool, thanks for doing this. 👍

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/utils/TestEventUtils.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+import com.palantir.atlasdb.jepsen.events.ImmutableOkEvent;
+import com.palantir.atlasdb.jepsen.events.OkEvent;
+import com.palantir.atlasdb.jepsen.events.RequestType;
+
+public abstract class TestEventUtils {
+
+    private static final String LOCKNAME = "default_lockname";
+    private static final String TIMESTAMP = "timestamp";
+
+    public static ImmutableInvokeEvent invokeLock(long time, int process, String lockname) {
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/utils/TestEventUtils.java]: Generally we prefer methods to return the interfaces (so `InvokeEvent`, etc.) as opposed to (`ImmutableInvokeEvent` etc.). There isn't any specific reason why we want the events created to be from Immutables.

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/utils/TestEventUtils.java [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+    public static ImmutableInfoEvent createInfoEvent(long time, String process, String requestType, String value) {
+        return ImmutableInfoEvent.builder()
+                .time(time)
+                .process(process)
+                .function(requestType)
+                .value(value)
+                .build();
+    }
+}
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/utils/TestEventUtils.java]: Thanks a lot for doing this class, it will help a lot with future Jepsen work 👍

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java [hunk 9]
```diff
@@ -32,15 +33,35 @@
...
+
+    public boolean isFailure() {
+        return value() == null
+                || (function().equals(RequestType.UNLOCK) && value().equals(UNLOCK_FAILURE))
+                || value().equals(FAILURE);
+    }
+
+    public boolean isSuccessful() {
+        return value().equals(SUCCESS)
+                || (function().equals(RequestType.UNLOCK) && value().equals(UNLOCK_SUCCESS));
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java]: Looking at this the two aren't generally complementary (e.g. an unlock which has some other value?), but would it be enough for our checkers to assume !failure is success?

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java [hunk 10]
```diff
@@ -32,15 +33,35 @@
...
         visitor.visit(this);
     }
+
+    public boolean isFailure() {
+        return value() == null
+                || (function().equals(RequestType.UNLOCK) && value().equals(UNLOCK_FAILURE))
+                || value().equals(FAILURE);
+    }
+
+    public boolean isSuccessful() {
```
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java]: It was before, but I figured that if you get something undefined, you want to return false on both accounts

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java [hunk 11]
```diff
@@ -32,15 +33,35 @@
...
 
+    @JsonProperty("f")
+    public abstract String function();
+
     @Override
     public void accept(EventVisitor visitor) {
         visitor.visit(this);
     }
+
+    public boolean isFailure() {
```
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/events/OkEvent.java]: That makes sense

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+         *  - FAILURE: was not holding the lock.
+         *
+         * REFRESH:
+         *  - SUCCESS: was holding the lock, still possibly does.
+         *  - FAILURE: was not holding the lock.
+         */
+        @Override
+        public void visit(OkEvent event) {
+            int currentProcess = event.process();
+            String lockName = resourceName.get(currentProcess);
```
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java]: We can do that, but I thought it cannot happen by design of the jepsen tests. Once we start going down this route, we also have to consider multiple consecutive InvokeEvents by the same process!

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+                case RequestType.UNLOCK:
+                    if (event.isSuccessful()) {
+                        if (!refreshAllowed.get(processLock)) {
+                            errors.add(lastEvent.get(processLock));
+                            errors.add(event);
+                        }
+                    }
+                    refreshAllowed.put(processLock, false);
+                    break;
+                default: break;
```
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java]: I believe findBugs or checkStyle complained before I added it, but will double-check

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+                locksHeld.put(lockName, TreeRangeSet.create());
+                locksAtSomePoint.put(lockName, new ArrayList<>());
+            }
+        }
+
+        @Override
+        public void visit(OkEvent event) {
+            Integer process = event.process();
+            String lockName = resourceName.get(process);
+            Pair processLock = new Pair(process, lockName);
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: As before, prefer parameterised types to raw types

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+            InvokeEvent invokeEvent = pendingForProcessAndLock.get(processLock);
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: I think this assumes that an ok event must be preceded by the appropriate invoke event. Probably reasonable, though it would be nice to throw something a bit more helpful

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+
+            switch (event.function()) {
+                /**
+                 * Successful LOCK:
+                 * 1) Add a new uncertain interval to verify at the end,
+                 * 2) Remember the new value for the most recent successful lock
+                 */
+                case RequestType.LOCK:
+                    if (event.isSuccessful()) {
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: In general this checker doesn't make any guarantees about failed events. It seems that we could just return immediately if the event isn't successful

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        @Override
+        public void visit(FailEvent event) {
+        }
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: (as before, not needed - default impl does nothing)

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+ * [a, b], where
+ * a is the InvokeEvent.time() of the lock request, and
+ * b is its OkEvent.time()
+ *
+ * We mantain a set L of all intervals where we know there was a lock, and a separate list U of all intervals where
+ * there was a lock at some point. In the latter list, we purposefully do not merge intervals, to ensure no information
+ * is lost.
+ *
+ * To verify locks were correctly granted, it is necessary and sufficient to verify that none of the uncertain
+ * intervals in U are fully covered by the intervals in the set of lock intervals L.
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: nit: I think it is necessary, but *not* sufficient, because of the cases of events occurring at a resolution below what our time allows (I assume this is possible since we have the small window and interval edge case tests). For example, suppose we have 2 processes both do invoke/ok for lock and refresh all at the "same time". We could have the actual sequence of events be [1 lock, 1 lock success, 2 lock, 2 lock success, 1 refresh, 1 refresh success, 2 refresh, 2 refresh success] which is not allowed. We're looking to flag cases which are definitely wrong, but for those which are possibly right and possibly wrong we play defensively and let it go.

I *think* it is necessary and sufficient for the case where all events have unique timestamps, though.

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+ */
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: Great job! 👍  At some point I think it would be nice to put something like this into the docs.

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java (hunk 13 above)
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java]: I'd say yes, and in fact we should throw `IllegalStateException` or something else in this case. This will help if we say try to feed a timestamp history into the lock checker - we'll currently get valid since there are no meaningful events 😬

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+                    break;
+                case RequestType.REFRESH:
+                    if (event.isFailure()) {
+                        refreshAllowed.put(processLock, false);
+                    }
+                    if (event.isSuccessful()) {
+                        if (!refreshAllowed.get(processLock)) {
+                            errors.add(lastEvent.get(processLock));
+                            errors.add(event);
+                            refreshAllowed.put(processLock, true);
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java]: 👍 Yep. We can combine the if-conditions as well.

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+            }
+
+            switch (event.function()) {
+                case RequestType.LOCK:
+                    if (event.isFailure()) {
+                        refreshAllowed.put(processLock, false);
+                    }
+                    if (event.isSuccessful()) {
+                        refreshAllowed.put(processLock, true);
+                    }
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessChecker.java]: If it's reasonable to assume these are complements (see other comment), then this could be `refreshAllowed.put(processLock, event.isSuccessful())`

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+                        lastHeldLock.put(processLock, event);
+                    }
+                    break;
+                /**
+                 * Successful REFRESH/UNLOCK:
+                 * Add the new interval [a, b) to the set of known locks, where
+                 * a is the OkEvent.time() of the last successful lock or the InvokeEvent.time() of refresh, and
+                 * b is the InvokeEvent.time() of the current request, if and only if b > a.
+                 * Also verify that the whole interval was free. Unlock can be treated as refresh, as the correctness
+                 * of their mutual interaction is verified by IsolatedProcessCorrectnessChecker
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java]: Some thoughts on this:

- offhand, it seems a bit strange that we have `[a, b)` here but `(a, b)` in the `LockCorrectnessChecker`. There are probably reasons why this is safe (something to do with the `b > a`?) though
- would be good if we document how we pick "the last successful lock or the InvokeEvent.time() of refresh" (if I read this correctly, the first refresh adds the lock time whilst subsequent refreshes add the last refresh's invoke time)

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java [hunk 23]
```diff
@@ -0,0 +1,147 @@
...
+                case RequestType.UNLOCK:
+                    if (event.isSuccessful() && lastHeldLock.containsKey(processLock)) {
+                        long lastLockTime = lastHeldLock.get(processLock).time();
+                        if (lastLockTime < invokeEvent.time()) {
+                            Range<Long> newRange = Range.closedOpen(lastLockTime, invokeEvent.time());
+                            if (!locksHeld.get(lockName).complement().encloses(newRange)) {
+                                log.error("A {} request for lock {} by process {} invoked at time {} was granted at "
+                                        + "time {}, but another process was granted the lock between {} and {} "
+                                        + "(last known time the lock was held by {})",
+                                        invokeEvent.function(), invokeEvent.value(), invokeEvent.process(),
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java]: I think the first 4 substitutions here should be `event` (dealing with the current event that was successful when it shouldn't have been), not `invokeEvent`.

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java [hunk 24]
```diff
@@ -0,0 +1,147 @@
...
+                 * Also verify that the whole interval was free. Unlock can be treated as refresh, as the correctness
+                 * of their mutual interaction is verified by IsolatedProcessCorrectnessChecker
+                 */
+                case RequestType.REFRESH:
+                case RequestType.UNLOCK:
+                    if (event.isSuccessful() && lastHeldLock.containsKey(processLock)) {
+                        long lastLockTime = lastHeldLock.get(processLock).time();
+                        if (lastLockTime < invokeEvent.time()) {
+                            Range<Long> newRange = Range.closedOpen(lastLockTime, invokeEvent.time());
+                            if (!locksHeld.get(lockName).complement().encloses(newRange)) {
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java]: Hmm... if I'm not wrong: the complement encloses newRange iff the intersection is zero. So I'd prefer to write this as `!locksHeld.get(lockName).subRangeSet(newRange).isEmpty()` (or maybe even factor out a separate method for this).

(Generally I find double negations more difficult to reason about, so usually if we can avoid them that's a good thing!)

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java [hunk 25]
```diff
@@ -0,0 +1,150 @@
...
+                .add(TestEventUtils.invokeLock(1, process2))
+                .add(TestEventUtils.lockSuccess(2, process2))
+                .add(TestEventUtils.invokeLock(3, process3))
+                .add(TestEventUtils.lockSuccess(4, process3))
+                .add(TestEventUtils.invokeRefresh(4, process1))
+                .add(TestEventUtils.refreshSuccess(5, process1))
+                .build();
+        CheckerResult result = runLockCorrectnessChecker(eventList);
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
```
Comment (jeremyk-91) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java]: This is a good test, but we should hopefully give a brief comment or two explaining _why_ the expected behaviour is that this is correct (i.e. you can have 2 get the lock, drop it, 1 get the lock, and 1 refresh just before it's lost and 3 gets the lock after that).

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java (hunk 4 above)
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java]: There is no previous OkEvent that is the cause of the problem. I replaced it with the InvokeEvent now, but not sure this is actually a nicer solution...

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java (hunk 5 above)
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java]: Good point! I also added cannotRefreshWithWrongLock to IsolatedProcessCorrectnessCheckerTest for the same reason 👍

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java (hunk 24 above)
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java]: Makes sense

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java (hunk 23 above)
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java]: event doesn't have all of that data, in particular the lock name, since the OkEvent puts the token instead of the lock name in value(). And then also giving both the invoke and the ok times for debugging

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java (hunk 18 above)
Comment (gmaretic) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessChecker.java]: You are right, we also do not actually know if the locks were held longer, possibly up to the point of an unsuccessful refresh attempt!

**Comment by gmaretic** [comment]
Should be all good, a couple of points open for discussion, but once that is done, ready to merge
